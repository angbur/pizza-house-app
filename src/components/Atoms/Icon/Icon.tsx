const icons = [
  'user',
  'menu',
  'order',
  'email',
  'basket',
  'facebook',
  'instagram',
  'arrow-down',
  'arrow-up',
  'vege',
  'meat',
  'seafood',
  'spicy',
  'pizza-slice',
  'pizza-circle',
  'cancel-cross',
  'trash',
  'circle',
  'list-icon',
  'info',
  'login',
  'menu-burger',
  'filter',
  'people',
  'plus',
  'minus',
];
export type IconName = typeof icons[number];

type IconProps = {
  name: IconName;
  size: number;
  isActive?: boolean;
};
const Icon = ({ name, size, isActive = false }: IconProps) => {
  return isActive ? (
    <img src={require(`assets/icon/${name}-active.svg`)} height={size} />
  ) : (
    <img src={require(`assets/icon/${name}.svg`)} height={size} />
  );
};

export default Icon;
