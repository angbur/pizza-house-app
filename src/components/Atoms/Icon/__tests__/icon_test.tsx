import renderer from 'react-test-renderer';
import ThemeProvider from 'components/Theme/ThemeContext';
import Icon from '../Icon';

const Wrapper = () => {
  return <ThemeProvider>{Icon({ name: 'menu-burger', size: 20, isActive: false })}</ThemeProvider>;
};

describe('Icon', () => {
  it('should renders correctly icon', () => {
    const tree = renderer.create(<Wrapper />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
