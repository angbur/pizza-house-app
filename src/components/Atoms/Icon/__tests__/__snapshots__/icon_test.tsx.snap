// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should renders correctly icon 1`] = `
<img
  height={20}
  src={
    Object {
      "ReactComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "render": [Function],
      },
      "__esModule": true,
      "default": "menu-burger.svg",
    }
  }
/>
`;
